backend/core/src/main/java/org/sonarsource/sonarlint/core/BindingCandidatesFinder.java

```diff
@@ -50,48 +51,54 @@ public BindingCandidatesFinder(ConfigurationRepository configRepository, Binding
     this.sonarProjectsCache = sonarProjectsCache;
   }
 
-  public Set<String> findConfigScopesToBind(String connectionId, String projectKey, SonarLintCancelMonitor cancelMonitor) {
+  public Set<ConfigurationScopeSharedContext> findConfigScopesToBind(String connectionId, String projectKey, SonarLintCancelMonitor cancelMonitor) {
     var configScopeCandidates = configRepository.getAllBindbableUnboundScopes();
     if (configScopeCandidates.isEmpty()) {
       return Set.of();
     }
 
-    var goodConfigScopeCandidates = new HashSet<ConfigurationScope>();
+    var goodConfigScopeCandidates = new HashSet<ConfigurationScopeSharedContext>();
 
     for (var scope : configScopeCandidates) {
-      cancelMonitor.checkCanceled();
-      if (checkIfScopeIsGoodCandidateForBinding(scope, connectionId, projectKey, cancelMonitor)) {
-        goodConfigScopeCandidates.add(scope);
-      }
+      checkIfScopeIsGoodCandidateForBinding(scope, connectionId, projectKey, cancelMonitor).ifPresent(goodConfigScopeCandidates::add);
     }
 
     // if both a parent and a child configuration scope are candidates, preference should be given to the higher scope in the hierarchy
     // we prefer to bind at the broadest possible scope
     return filterOutLeafCandidates(goodConfigScopeCandidates);
   }
 
-  private static Set<String> filterOutLeafCandidates(Set<ConfigurationScope> candidates) {
-    var candidateIds = candidates.stream().map(ConfigurationScope::getId).collect(Collectors.toSet());
-    return candidates.stream().filter(scope -> {
-      var parentId = scope.getParentId();
-      return parentId == null || !candidateIds.contains(parentId);
-    }).map(ConfigurationScope::getId).collect(Collectors.toSet());
-  }
+  private Optional<ConfigurationScopeSharedContext> checkIfScopeIsGoodCandidateForBinding(
+    ConfigurationScope scope, String connectionId, String projectKey, SonarLintCancelMonitor cancelMonitor) {
+    cancelMonitor.checkCanceled();
 
-  private boolean checkIfScopeIsGoodCandidateForBinding(ConfigurationScope scope, String connectionId, String projectKey, SonarLintCancelMonitor cancelMonitor) {
     var cluesAndConnections = bindingClueProvider.collectBindingCluesWithConnections(scope.getId(), Set.of(connectionId), cancelMonitor);
 
     var cluesWithMatchingProjectKey = cluesAndConnections.stream()
       .filter(c -> projectKey.equals(c.getBindingClue().getSonarProjectKey()))
       .collect(toList());
+
+
     if (!cluesWithMatchingProjectKey.isEmpty()) {
-      return true;
+      var isFromSharedConfiguration = cluesWithMatchingProjectKey.stream()
+        .anyMatch(c -> Boolean.TRUE == c.getBindingClue().isFromSharedConfiguration());
```

Comment:
can be simplified to `.anyMatch(c -> c.getBindingClue().isFromSharedConfiguration());`